### PR TITLE
Page.select Calling mainframe.select() Fixes #1139 

### LINKED ISSUE #1139: Missing ElementHandle.select; impossible (?) to make selection in a frame
There is no `frame.select` equivalent of the [`page.select`](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageselectselector-values). As far as I can tell, there is no way to select an option within an iframe.

### PR DESCRIPTION
**Pull request by suvratjain1995**

### CODE REVIEW

**Comment by aslushnikov** [comment]
@suvratjain1995 are you the author of #1198? If yes, should the #1198 be closed in favor of this PR?

**Comment by suvratjain1995** [comment]
@aslushnikov yes I am author that ID was having some issue regarding cla and all

File: docs/api.md [hunk 1]
```diff
@@ -1507,6 +1508,19 @@ If the name is empty, returns the id attribute instead.
 #### frame.parentFrame()
 - returns: <[Frame]> Returns parent frame, if any. Detached frames and main frames return `null`.
 
+#### frame.select(selector, ...values)
+- `selector` <[string]> A [selector] to query page for
```
Comment (aslushnikov) [inline docs/api.md]: nit: s/page/frame

File: test/test.js [hunk 2]
```diff
@@ -901,7 +901,7 @@ describe('Page', function() {
       await page.goto('asdfasdf').catch(e => error = e);
       expect(error.message).toContain('Cannot navigate to invalid URL');
     }));
-    it('should fail when navigating to bad SSL', SX(async function() {
+    xit('should fail when navigating to bad SSL', SX(async function() {
```
Comment (aslushnikov) [inline test/test.js]: not need to skip this any more

File: lib/Page.js [hunk 3]
```diff
@@ -782,17 +782,7 @@ class Page extends EventEmitter {
...
-        throw new Error('Element is not a <select> element.');
-      const options = Array.from(element.options);
-      element.value = undefined;
-      for (const option of options)
-        option.selected = values.includes(option.value);
-      element.dispatchEvent(new Event('input', { 'bubbles': true }));
-      element.dispatchEvent(new Event('change', { 'bubbles': true }));
-      return options.filter(option => option.selected).map(option => option.value);
-    }, values);
+    return this.mainFrame().select(selector, ...values);
```
Comment (aslushnikov) [inline lib/Page.js:785]: can you please state in `page.select` documentation that it's a shortcut for `page.mainFrame().select`.

E.g. see how it's done for [page.$](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pageselector)

**Review by aslushnikov — COMMENTED** [review]
Overall looks good, just a few nits

File: docs/api.md (hunk 1 above)
Comment (suvratjain1995) [reply docs/api.md]: @aslushnikov I couldnt get you here.

Comment (aslushnikov) [reply docs/api.md]: @suvratjain1995 sorry, I meant that "frame" should be used instead of "page".

File: lib/FrameManager.js [hunk 4]
```diff
@@ -407,6 +407,25 @@ class Frame {
   }
 
   /**
+  *@param {string} selector
+  *@param {!Array<string>} values
```
Comment (aslushnikov) [inline lib/FrameManager.js]: 1. there should be spaces between "*" and "@"
2. this should also have a @return annotation:

```js
/**
  ...
  * @return {!Promise<!Array<string>>}
 */
```

**Comment by suvratjain1995** [comment]
@aslushnikov worked out those .

**Review by aslushnikov — APPROVED** [review]
Awesome, thanks!

**Pull request merged**